src/Workspaces/VisualBasic/Portable/Simplification/VisualBasicInferredMemberNameReducer.Rewriter.vb

```diff
@@ -0,0 +1,44 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Threading
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Simplification
+    Partial Friend Class VisualBasicInferredMemberNameReducer
+
+        Private Class Rewriter
+            Inherits AbstractExpressionRewriter
+
+            Public Sub New(optionSet As OptionSet, cancellationToken As CancellationToken)
+                MyBase.New(optionSet, cancellationToken)
+            End Sub
+
+            Public Overrides Function VisitSimpleArgument(node As SimpleArgumentSyntax) As SyntaxNode
+                CancellationToken.ThrowIfCancellationRequested()
+
+                Dim newNode = MyBase.VisitSimpleArgument(node)
+
+                If node.IsParentKind(SyntaxKind.TupleExpression) Then
+                    Return SimplifyExpression(
+                        node,
+                        newNode:=newNode,
+                        simplifier:=AddressOf SimplifyTupleName)
+                End If
+
+                Return newNode
+            End Function
+
+            Public Overrides Function VisitNamedFieldInitializer(node As NamedFieldInitializerSyntax) As SyntaxNode
+                Dim newNode = MyBase.VisitNamedFieldInitializer(node)
+
+                Return SimplifyExpression(
+                    node,
+                    newNode:=newNode,
+                    simplifier:=AddressOf SimplifyNamedFieldInitializer)
+            End Function
+
```

Comment:
nit: in general, i do not like extraneous newline between the 'end' statements of VB constructs.